packages/shared_preferences/shared_preferences_foundation/darwin/Classes/BetterSharedPreferencesPlugin.swift

```diff
@@ -0,0 +1,131 @@
+    return getAllPrefs(allowList: allowList)
+  }
+
+  func setBool(key: String, value: Bool, options: SharedPreferencesPigeonOptions) {
+    getUserDefaults(options: options).set(value, forKey: key)
+  }
+
+  func setDouble(key: String, value: Double, options: SharedPreferencesPigeonOptions) {
+    getUserDefaults(options: options).set(value, forKey: key)
+  }
+
+  func setValue(key: String, value: Any, options: SharedPreferencesPigeonOptions) {
+    getUserDefaults(options: options).set(value, forKey: key)
+  }
+
+  func getString(key: String, options: SharedPreferencesPigeonOptions) throws -> String? {
+    return getUserDefaults(options: options).string(forKey: key)
+  }
+  
+  func getBool(key: String, options: SharedPreferencesPigeonOptions) throws -> Bool? {
+    getUserDefaults(options: options).bool(forKey: key)
+  }
+  
+  func getDouble(key: String, options: SharedPreferencesPigeonOptions) throws -> Double? {
+    getUserDefaults(options: options).double(forKey: key)
+  }
+  
+  func getInt(key: String, options: SharedPreferencesPigeonOptions) throws -> Int64? {
+    getUserDefaults(options: options).integer(forKey: key)
+  }
+  
+  func getStringList(key: String, options: SharedPreferencesPigeonOptions) throws -> [String]? {
+    getUserDefaults(options: options).stringArray(forKey: key)
+  }
+
+  func remove(key: String, options: SharedPreferencesPigeonOptions) {
+    getUserDefaults(options: options).removeObject(forKey: key)
+  }
+
+  func clear(allowList: [String]?, options: SharedPreferencesPigeonOptions) -> Bool {
+    let defaults = getUserDefaults(options: options)
+    for (key, _) in getAllPrefs(allowList: allowList) {
+      defaults.removeObject(forKey: key)
+    }
+    return true
+  }
+
+  /// Returns all preferences stored with specified prefix.
+  /// If [allowList] is included, only items included will be returned.
+  /// If no [allowList], returns supported types only.
+  func getAllPrefs(allowList: [String]?, options: SharedPreferencesPigeonOptions) -> [String: Any] {
+    var filteredPrefs: [String: Any] = [:]
+    var allowSet: Set<String>?
+    if let allowList = allowList {
+      allowSet = Set(allowList)
+    }
+    if let appDomain = Bundle.main.bundleIdentifier,
+      let prefs = getUserDefaults(options: options).persistentDomain(forName: appDomain)
+    {
+      if allowList == nil {
+        for (key, value) in prefs where (isTypeCompatible(value: value)) {
+          filteredPrefs[key] = value
+        }
+      } else {
+        for (key, value) in prefs where (allowSet!.contains(key)) {
+          filteredPrefs[key] = value
+        }
+      }
+    }
+    return filteredPrefs
+  }
+
+  func isTypeCompatible(value: Object) -> Bool {
+    if let value = bool as? Bool {
+      return true
+    }
+    if let value = double as? Double {
+      return true
+    }
+    if let value = string as? String {
+      return true
+    }
+    if let value = int as? Int {
+      return true
```

Comment:
nit: can check the runtime type

```
switch type(of: value) {
  ...
}
```

Reply:
I've been trying this to no avail, can you expand further?